test/lib/bamboo/test_adapter_test.exs

```diff
@@ -44,20 +44,107 @@ defmodule Bamboo.TestAdapterTest do
 
     assert_delivered_email sent_email
     refute_delivered_email unsent_email
+
+    sent_email |> TestMailer.deliver_now
+    assert_raise ExUnit.AssertionError, fn ->
+      assert_delivered_email %{sent_email | to: "oops"}
+    end
+  end
+
+  test "assert_delivered_email with no delivered emails" do
+    sent_email = new_email(from: "foo@bar.com", to: ["foo@bar.com"])
+
+    try do
+      assert_delivered_email %{sent_email | to: "oops"}
+    rescue
+      error in [ExUnit.AssertionError] ->
```

Comment:
This is some funky stuff. I got it from https://github.com/elixir-lang/elixir/blob/master/lib/ex_unit/test/ex_unit/assertions_test.exs#L591

It's how they test their assertion helpers in ExUnit